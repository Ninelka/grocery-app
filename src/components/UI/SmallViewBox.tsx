import { StyleSheet, Text, View } from 'react-native';
import Link from './Link';
import { ReactNode } from 'react';
import { COLORS, FONT_FAMILY, GlobalStyles } from '../../constants';

interface ISmallViewBox {
  onSeeAll: () => void;
  children: ReactNode;
  title?: string;
}

function SmallViewBox({ title, onSeeAll, children }: ISmallViewBox) {
  return (
    <View>
      <View style={styles.header}>
        <Text style={styles.title}>{title}</Text>
        <Link onPress={onSeeAll}>See All</Link>
      </View>
      <View>{children}</View>
    </View>
  );
}

export default SmallViewBox;

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: GlobalStyles.spacing.s,
  },
  title: {
    fontFamily: FONT_FAMILY.bold,
    fontSize: GlobalStyles.fontSize.title1,
    color: COLORS.labelsPrimary,
  },
});
