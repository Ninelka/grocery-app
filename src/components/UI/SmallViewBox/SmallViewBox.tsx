import { StyleSheet, Text, View } from 'react-native';
import Link from '../Link/Link';
import { ReactNode } from 'react';
import { COLORS, FONT_FAMILY, GlobalStyles } from '../../../constants';

interface ISmallViewBox {
  onSeeAll: () => void;
  children: ReactNode;
  title?: string;
}

function SmallViewBox({ title, onSeeAll, children }: ISmallViewBox) {
  return (
    <View testID="small-view-box-container">
      <View style={styles.header}>
        <Text testID="small-view-box-title" style={styles.title}>
          {title}
        </Text>
        <Link onPress={onSeeAll}>See All</Link>
      </View>
      <View>{children}</View>
    </View>
  );
}

export default SmallViewBox;

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: GlobalStyles.spacing.s,
  },
  title: {
    fontFamily: FONT_FAMILY.bold,
    fontSize: GlobalStyles.fontSize.title1,
    color: COLORS.labelsPrimary,
  },
});
