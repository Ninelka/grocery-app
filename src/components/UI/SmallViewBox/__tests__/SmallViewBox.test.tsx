import SmallViewBox from '../SmallViewBox';
import { render, screen } from '@testing-library/react-native';
import { Text } from 'react-native';

const onPressSeeAllBtn = jest.fn();

describe('SmallViewBox', function () {
  beforeEach(() => {
    render(
      <SmallViewBox title="Test title" onSeeAll={onPressSeeAllBtn}>
        <Text>No any data</Text>
      </SmallViewBox>
    );
  });
  test('renders component', () => {
    expect(screen).toMatchSnapshot();
  });

  test('should contain 2 children', function () {
    expect(screen.getByTestId('small-view-box-container').children.length).toBe(
      2
    );
  });

  test('should show the title prop', () => {
    expect(screen.getByTestId('small-view-box-title').children).toEqual([
      'Test title',
    ]);
  });
});
