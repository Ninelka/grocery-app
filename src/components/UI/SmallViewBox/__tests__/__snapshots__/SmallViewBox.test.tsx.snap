// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallViewBox renders component 1`] = `
<View
  testID="small-view-box-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#22292e",
          "fontFamily": "nunito-bold",
          "fontSize": 24,
        }
      }
      testID="small-view-box-title"
    >
      Test title
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
        ]
      }
      testID="link-btn-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#54b175",
              "fontFamily": "nunito-bold",
              "fontSize": 17,
            },
            undefined,
          ]
        }
        testID="link-btn-text"
      >
        See All
      </Text>
    </View>
  </View>
  <View>
    <Text>
      No any data
    </Text>
  </View>
</View>
`;
